Samples/Mutation_Files/crossover.py

```diff
@@ -0,0 +1,37 @@
+import random
+
+def single_point_crossover(timetable1, timetable2):
+    # Ensure both parents have the same length
+    assert len(timetable1) == len(timetable2), "Parents must have the same length"
+    
+    # Choose a random crossover point
+    crossover_point = random.randint(1, len(timetable1) -1)
+    
+    # Generate offspring by swapping genes at the crossover point
+    offspring1 = timetable1[:crossover_point] + timetable2[crossover_point:]
+    offspring2 = timetable2[:crossover_point] + timetable1[crossover_point:]
+    
+    return offspring1, offspring2
+
+# Example parents with dictionaries representing class schedules
+timetable1 = [
+    {"section": "A", "teacher": "sashi kumar", "time_slot": 1, "classRoom": "LT01"},
+    {"section": "A", "teacher": "deepa nair", "time_slot": 2, "classRoom": "LT03"},
+    {"section":"B", "teacher": "subhankar goyal", "time_slot": 2,"classRoom":"LT01"},
+    {"section":"A", "teacher": "jeet ", "time_slot": 3,"classRoom":"LT03"},
+]
+
+timetable2 = [
+    {"section": "A", "teacher": "mehul manu", "time_slot": 3, "classRoom": "LT02"},
+    {"section": "A", "teacher": "priya sharma", "time_slot": 4, "classRoom": "LT04"},
+    {"section":"C", "teacher": "kiran kher", "time_slot": 4,"classRoom":"LT02"},
+    {"section":"B", "teacher": "devender", "time_slot": 4,"classRoom":"LT02"}
+]
+
```

Comment:
Hard and soft constraints are not included so there are conflicts where the same time slot is provided to 2 section